src/test/java/org/springframework/data/elasticsearch/core/ReactiveElasticsearchTemplateIntegrationTests.java

```diff
@@ -1188,6 +1190,45 @@ void shouldReturnMonoOfReactiveSearchHits() {
 				.verifyComplete();
 	}
 
+	@Test // #1529
+	void shouldWorkReindexForExistingIndex() {
+		String sourceIndexName = indexNameProvider.indexName();
+		String documentId = nextIdAsString();
+		ElasticsearchTemplateTests.SampleEntity sampleEntity = ElasticsearchTemplateTests.SampleEntity.builder().id(documentId).message("abc").build();
+		operations.save(sampleEntity).block();
+
+		indexNameProvider.increment();
+		String destIndexName = indexNameProvider.indexName();
+		operations.indexOps(IndexCoordinates.of(destIndexName)).create();
+		final PostReindexRequest postReindexRequest = PostReindexRequest
+				.builder(IndexCoordinates.of(sourceIndexName), IndexCoordinates.of(destIndexName))
+				.withRefresh(true)
+				.build();
+		operations.reindex(postReindexRequest)
+				.as(StepVerifier::create)
+				.consumeNextWith(postReindexResponse -> assertThat(postReindexResponse.getTotal()).isEqualTo(1L))
+				.verifyComplete();
+		NativeSearchQuery searchQuery = new NativeSearchQueryBuilder().withQuery(matchAllQuery()).build();
```

Comment:
```suggestion
    Query searchQuery = operations.matchAllQuery();
```
independent of the client implementation, helps when we migrate this to support the new client.